### ISSUE TITLE
DetCon Transform: Allow passing `Callable` for unsupervised segmentation

### ISSUE THREAD

**Issue by liopeer** [lightly-ai/lightly#1972]
Currently, the use of unsupervised masks with the DetConS (or DetConB) transforms requires constructing a helper function as the actual transform, see #1969 and #1971 . It would be much more elegant if the transforms could accept an optional `Callable[[PILImage | Tensor], PILImage | Tensor]` that constructs the mask.

The priority would be compatibility with unsupervised segmentation algorithms from scikit-image or cv2, such as Felzenszwalb, SLIC, Watershed etc.

**Comment by liopeer** [comment]
@saud5150 could be a good follow-up for you I think!

**Comment by saud5150** [comment]
Happy to take this.

**Comment by saud5150** [comment]
Thanks @liopeer! Picking this up. The example already ships the `try/except felzenszwalb` grid-fallback path from #1969, so this is the "pass a callable so it runs in the dataloader workers" part you flagged for later.

Plan for `DetConSTransform`:

```python
from functools import partial
from skimage.segmentation import felzenszwalb

transform = DetConSTransform(
    mask_fn=partial(felzenszwalb, scale=100, sigma=0.5, min_size=20),
    input_size=96,
)
```

- **`mask_fn: Callable[[Image], Tensor | ndarray]`** — receives the image, returns raw integer labels. It accepts skimage/cv2 segmenters (Felzenszwalb, SLIC, Watershed) drop-in and wraps the output into a `Mask` internally, so there's no user-side `Mask(...)` boilerplate and the whole thing composes inside the v2 transform pipeline (i.e. runs in the dataloader workers, as you noted).
- **Backward compatible** — purely additive; the `grid_size` and pre-segmented-`Mask` paths are untouched. `grid_size` and `mask_fn` are mutually exclusive (passing both raises `ValueError`).
- **Scope** — lands on `DetConSTransform`, the shared DetCon augmentation/masking transform (there's no separate `DetConBTransform`), so it covers both variants.
- I'll simplify `examples/pytorch/detcon.py` to use `mask_fn`, dropping the hand-rolled mask helper, and regenerate the notebook.

Wrapping the raw labels internally matches your `felzenszwalb` sketch — shout if you'd rather `mask_fn` return a `Mask` directly and I'll adjust.

**Comment by liopeer** [comment]
Looks like the right approach! One thing to consider is the typing of `mask_fn`, since imo it should be `Callable[[PILImage | Tensor], PILImage | Tensor]`, i.e. I think it should mainly support the types that the v2 torchvision transforms also support. From what I have seen you will have to override the transforms `forward` method.
